src/addBusinessDays/index.ts

```diff
@@ -42,49 +42,70 @@ export default function addBusinessDays(
       ? [1, 2, 3, 4, 5]
       : options.businessDays.filter((number) => number < 7).map(toInteger)
 
-  const date = toDate(dirtyDate)
+  const initialDate = toDate(dirtyDate)
   const amount = toInteger(dirtyAmount)
   let accountedForExceptions = 0
+
   const isExcepted = (date: Date): boolean | null => {
     if (options.exceptions) {
       const exception =
-        Object.keys(options.exceptions).find((e) =>
-          isSameDay(new Date(e), date)
-        ) || ''
-      if (options.exceptions[exception] !== undefined) {
+        Object.keys(exceptions).find((e) => isSameDay(new Date(e), date)) || ''
+      if (exceptions[exception] === true) {
+        accountedForExceptions -= 1
+      } else if (exceptions[exception] === false) {
         accountedForExceptions += 1
       }
-      return options.exceptions[exception]
+      return exceptions[exception]
     }
     return null
   }
-  const isNonWorkingDay = (date: Date) =>
-    isExcepted(date) === false || !businessDays.includes(date.getDay())
+
+  const isWorkingDay = (date: Date) => {
+    const hasException = isExcepted(date)
+    if (hasException === false) return false
+    if (hasException === true || businessDays.includes(date.getDay())) {
+      return true
+    }
+    return false
+  }
 
   if (isNaN(amount)) return new Date(NaN)
 
-  if (date.toString() === 'Invalid Date') {
-    return date
+  if (initialDate.toString() === 'Invalid Date') {
+    return initialDate
   }
-  const startedOnNonWorkingDay = isNonWorkingDay(date)
-  const hours = date.getHours()
+
+  // We need to call this to add the initial date to the accountedForExceptions if it's an exception
+  // Ideally we shouldn't need to do this
+  isExcepted(initialDate)
+  const startedOnNonBusinessDay = !businessDays.includes(initialDate.getDay())
+  const hours = initialDate.getHours()
   const sign = amount < 0 ? -1 : 1
-  const fullWeeks = toInteger(amount / businessDays.length)
-  date.setDate(date.getDate() + fullWeeks * 7)
 
+  let newDate = new Date(initialDate)
+  let fullWeeks = toInteger(amount / businessDays.length)
   // Get remaining days not part of a full week
   let restDays = Math.abs(amount % businessDays.length)
+
+  // Fixes edge case bug to account for false exceptions in the first week
+  if (options.exceptions && restDays === 0 && fullWeeks === 1) {
+    restDays = businessDays.length
+    fullWeeks = 0
+  }
```

Comment:
thought: Let's discuss this when you are online. I'd like to understand the issue/rational further. I'm also a little concerned this might be pushing the bugs to the second week?